src/llscan.cc

```diff
@@ -0,0 +1,497 @@
+  if (type == v8->types()->kMapType) {
+    return heap_object.Inspect(&options, err);
+  }
+
+  if (type == v8->types()->kJSObjectType) {
+    v8::HeapObject map_obj = heap_object.GetMap(err);
+    if (err.Fail()) {
+      return std::string();
+    }
+
+    v8::Map map(map_obj);
+    v8::HeapObject constructor_obj = map.Constructor(err);
+    if (err.Fail()) {
+      return std::string();
+    }
+
+    int64_t constructor_type = constructor_obj.GetType(err);
+    if (err.Fail()) {
+      return std::string();
+    }
+
+    if (constructor_type != v8->types()->kJSFunctionType) {
+      return "<Object: no constructor>";
+    }
+
+    v8::JSFunction constructor(constructor_obj);
+
+    return constructor.Name(err);
+  }
+
+  if (type == v8->types()->kHeapNumberType) {
+    return "(HeapNumber)";
+  }
+
+  if (type == v8->types()->kJSArrayType) {
+    return "(Array)";
+  }
+
+  if (type == v8->types()->kOddballType) {
+    return "(Oddball)";
+  }
+
+  if (type == v8->types()->kJSFunctionType) {
+    return "(Function)";
+  }
+
+  if (type == v8->types()->kJSRegExpType) {
+    return "(RegExp)";
+  }
+
+  if (type < v8->types()->kFirstNonstringType) {
+    return "(String)";
+  }
+
+  if (type == v8->types()->kFixedArrayType) {
+    return "(FixedArray)";
+  }
+
+  if (type == v8->types()->kJSArrayBufferType) {
+    return "(ArrayBuffer)";
+  }
+
+  if (type == v8->types()->kJSTypedArrayType) {
+    return "(ArrayBufferView)";
+  }
+
+  if (type == v8->types()->kJSDateType) {
+    return "(Date)";
+  }
+
+  std::string unknown("unknown: ");
+
+  return unknown += std::to_string(type);
+}
+
+
+bool ListInstancesCmd::DoExecute(SBDebugger d, char** cmd,
+                                 SBCommandReturnObject& result) {
+  SBTarget target = d.GetSelectedTarget();
+  if (!target.IsValid()) {
+    result.SetError("No valid process, please start something\n");
+    return false;
+  }
+
+  v8::Value::InspectOptions inspect_options;
+
+  inspect_options.detailed = detailed_;
+
+  char** start = ParseInspectOptions(cmd, &inspect_options);
+
+  std::string full_cmd;
+  for (; start != nullptr && *start != nullptr; start++) full_cmd += *start;
+
+  SBExpressionOptions options;
+  SBValue value = target.EvaluateExpression(full_cmd.c_str(), options);
+  if (value.GetError().Fail()) {
+    SBStream desc;
+    if (!value.GetError().GetDescription(desc)) return false;
+    result.SetError(desc.GetData());
+    return false;
+  }
+
+  // Load V8 constants from postmortem data
+  llv8.Load(target);
+
+  v8::Value v8_mapaddr(&llv8, value.GetValueAsSigned());
+  v8::Error err;
+
+  if (err.Fail()) {
+    result.SetError("Failed to evaluate expression");
+    return false;
+  }
+
+  uint64_t mapaddr = value.GetValueAsUnsigned();
+  std::map<uint64_t, TypeRecord*>::iterator instance_it =
+      llscan.GetMapsToInstances().find(mapaddr);
+  if (instance_it != llscan.GetMapsToInstances().end()) {
```

Comment:
I don't know, lldb doesn't seem to do paging by default (unlike gdb) and doesn't seem to have an option to enable it so in general that's not the way an lldb command would work.
That said we may want to limit the output to make it readable later on but we could have options to only show the largest/most numerous types of objects rather than paging.
Can we pull this out and make it a future enhancement? :-)